src/map/skill.cpp

```diff
@@ -8223,6 +8283,17 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 			skill_castend_damage_id);
 		status_change_end(src, SC_HIDING, INVALID_TIMER);
 		break;
+		
+	case SOA_SOUL_GATHERING:
+		if(sd) {
+			int limit = 5;
+			if( pc_checkskill(sd, SP_SOULENERGY) )
+				limit += pc_checkskill(sd, SP_SOULENERGY) * 3;
```

Comment:
```suggestion
			limit += pc_checkskill(sd, SP_SOULENERGY) * 3;
```
pc_checkskill returns 0 if not learned, so can be done like this.